### PR TITLE
Initial commit 

### PR DESCRIPTION
**Pull request by marosset**
Initial commit of script to produce a lightweight image that can be used as the base for hostprocess container images

Signed-off-by: Mark Rossetti <marosset@microsoft.com>

### CODE REVIEW

**Comment by marosset** [comment]
Let me move the existing README.md into CONTRIBUTING.md.
I didn't realize a README.md was created automatically.

File: README.md [hunk 1]
```diff
@@ -1,33 +1,50 @@
...
+Instead use BuildKit or other tools.
+
+Example:
+
+#### Create a builder
+
+One time step
+
+```cmd
+docker buildx create --name img-builder --use --platform windows/and64
```
Comment (vitaliy-leschenko) [inline README.md:41]: typo: amd64 instead of and64

File: README.md [hunk 2]
```diff
@@ -1,33 +1,59 @@
...
+### Benefits
 
-## Contributing
+Using this image as a base for HostProcess containers has a few advantages over using other base images for Windows containers including:
 
-This project welcomes contributions and suggestions.  Most contributions require you to agree to a
-Contributor License Agreement (CLA) declaring that you have the right to, and actually do, grant us
-the rights to use your contribution. For details, visit https://cla.opensource.microsoft.com.
+- Size - This image is a few KB. Even the smallest official base image (NanoServer) is still a few hundred MB is size.
+- OS compatibility - HostProcess containers do not inherit the same [compatibility requirements](https://docs.microsoft.com/virtualization/windowscontainers/deploy-containers/version-compatibility) as Windows server containers and because of this it does not make sense to include all of the runtime / system binaries that make up the different base layers. Using this image allows for a single container image to be used on any Windows Server version which can greatly simplify container build processes.
```
Comment (jsturtevant) [inline README.md:12]: should we explicitly note that this will not work as base image for process isolation?

File: New-HostProcessBaseImage.ps1 [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+            "${layerHash}/layer.tar"
+        ]
+    }
+]
+"@ | Out-File  -FilePath "build\image\manifest.json" -Encoding ascii
+
+# Tar the image
+tar.exe  -cf "build\host-process-scratch.tar" -C "build\image" .
+
+# Output a file with the image hash so we can import/push the image from CI
```
Comment (jsturtevant) [inline New-HostProcessBaseImage.ps1:139]: looks like the outcome of this script is tar file and manifest files? do we have anything that pushes it to a registry now?

File: .github/workflows/build.yaml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+  workflow_dispatch:
+
+permissions:
+  contents: read
+  packages: write
```
Comment (jsturtevant) [inline .github/workflows/build.yaml:14]: is this needed since it isn't pushing?

File: README.md (hunk 2 above)
Comment (marosset) [reply README.md:12]: Yea, I can add a note for emphasis

File: .github/workflows/build.yaml (hunk 4 above)
Comment (marosset) [reply .github/workflows/build.yaml:14]: whoops, thought i removed this.

File: README.md (hunk 2 above)
Comment (marosset) [reply README.md:14]: Added (in ##Overview)

File: New-HostProcessBaseImage.ps1 (hunk 3 above)
Comment (marosset) [reply New-HostProcessBaseImage.ps1:139]: No. 
The GH action in my repo where this is based on does a docker load + push.
We can't use GH actions to push to mcr.microsoft.com so I intentionally left that out.
I can document the steps we use to load + push in a follow up PR (once we've got the image published)

Comment (marosset) [reply New-HostProcessBaseImage.ps1:139]: x-ref https://github.com/marosset/host-process-scratch-image/blob/94c662a6a7615d5a2fea1b6d22948df5d0c69010/.github/workflows/build.yaml#L30-L37

Comment (jsturtevant) [reply New-HostProcessBaseImage.ps1:139]: sounds good!

**Comment by jsturtevant** [comment]
lgtm

**Review by jsturtevant — APPROVED** [review]

**Comment by marosset** [comment]
That last push was me added a gpg signature to the commit.

**Pull request merged**